src/EnzymeBot.ts

```diff
@@ -111,42 +114,41 @@ export class EnzymeBot {
 
     const contract = new ComptrollerLib(comptroller, this.wallet);
     return contract.callOnExtension.args(integrationManager, IntegrationManagerActionId.CallOnIntegration, callArgs);
-
   }
 
   public async tradeAlgorithmically() {
     // get a random token
     const randomToken = await this.chooseRandomAsset();
 
     // if no random token return, or if the random token is a derivative that's not available on Uniswap
-    if (!randomToken || randomToken.derivativeType) {
+    if (!randomToken) {
       console.log("The Miner's Delight did not find an appropriate token to buy.");
       return;
     }
 
     // get your fund's holdings
-    const vaultHoldings = await this.getHoldings();
+    const holdings = this.vaultDetails.vault?.trackedAssets;
 
     // if you have no holdings, return
-    if (vaultHoldings.length === 0) {
+    if (!holdings || holdings.length === 0) {
       console.log('Your fund has no assets.');
       return;
     }
 
     // if your vault already holds the random token, return
-    if (vaultHoldings.map((holding) => holding?.id.toLowerCase()).includes(randomToken.id.toLowerCase())) {
+    if (holdings.map((holding) => holding.id.toLowerCase()).includes(randomToken.id.toLowerCase())) {
       console.log("You already hold the asset that the Miner's Delight randomly selected.");
       return;
     }
 
     // get the amount of each holding
     const holdingAmounts = await Promise.all(
-      vaultHoldings.map((holding) => getTokenBalance(this.vaultAddress, holding!.id, this.network))
+      holdings.map((holding) => getTokenBalance(this.vaultAddress, holding.id, this.network))
     );
 
     // combine holding token data with amounts
-    const holdingsWithAmounts = vaultHoldings.map((item, index) => {
-      return { ...item, amount: holdingAmounts[index] };
+    const holdingsWithAmounts = holdings.map((holding, index) => {
+      return { ...holding, amount: holdingAmounts[index] };
     });
 
     // find the token you will sell by searching for largest token holding
```

Comment:
You should make clear that this uses the token with the largest holding (token amount), but does not take into account the dollar value of that token holding. So if the vault holds a large amount of an almost worthless token, it will still trade that token.